File: perl/stathat.pl
use HTTP::Request::Common qw(POST);
use LWP::UserAgent;

sub stathat_post {
        my ($path, $params) = @_;
        my $ua = LWP::UserAgent->new;
        my $req = POST 'http://api.stathat.com/' . $path, $params;
        return $ua->request($req)->as_string;
};

sub stathat_count {
        my ($stat_key, $user_key, $count) = @_;
        return stathat_post('c', [ key => $stat_key, ukey => $user_key, count => $count ]);
};

sub stathat_value {
        my ($stat_key, $user_key, $value) = @_;
        return stathat_post('v', [ key => $stat_key, ukey => $user_key, value => $value ]);
};

sub stathat_ez_count {
        my ($email, $stat_name, $count) = @_;
        return stathat_post('ez', [ email => $email, stat => $stat_name, count => $count ]);
};

sub stathat_ez_value {
        my ($email, $stat_name, $value) = @_;
        return stathat_post('ez', [ email => $email, stat => $stat_name, value => $value ]);
};

File: lua/stathat.lua
-- can download this package at https://github.com/diegonehab/luasocket
-- or via luarocks: luarocks install luasocket
local http = require("socket.http")

local stathat = {}

function stathat.ez_count(ezkey, stat_name, count)
        return http.request("http://api.stathat.com/ez", ("ezkey=%s&stat=%s&count=%s"):format(ezkey, stat_name, count)
end

function stathat.ez_value(ezkey, stat_name, value)
        return http.request("http://api.stathat.com/ez", ("ezkey=%s&stat=%s&value=%s"):format(ezkey, stat_name, value)
end

function stathat.count(stat_key, user_key, count)
        return http.request("http://api.stathat.com/c", ("key=%s&ukey=%s&count=%s"):format(stat_key, user_key, count)
end

function stathat.value(stat_key, user_key, value)
        return http.request("http://api.stathat.com/v", ("key=%s&ukey=%s&value=%s"):format(stat_key, user_key, value)
end

return stathat
